src/analytics.ts

```diff
@@ -140,13 +139,37 @@ export const logEvent = ({
         value = DEFAULT_EVENT_VALUE;
     }
 
-    const params = {
+    return {
         eventCategory: category,
         eventAction: action,
         eventLabel: formatLabel(label),
         eventValue: value,
         ...fieldsObject,
     };
+};
+
+export const logEvent = (event: TrackingEvent): Promise<void> => {
+    let {name, ...params} = event;
+    if (!name) {
+        // If the event doesn't have a name, it's a legacy analytics event
+        if (!event.category || !event.action) {
+            console.warn(
+                'LegacyAnalyticsEvent should have "category" and "action"',
+                {
+                    category: event.category,
+                    action: event.action,
+                },
+            );
+            return Promise.resolve();
+        }
+        params = getLegacyAnalyticsEventParams(event as LegacyAnalyticsEvent);
+        name = event.category;
+    }
+
+    if (params.component_copy) {
+        // @ts-ignore - params is a new object created from event destructuring, so TS shouldn't complain about it being readonly.
+        params.component_copy = formatLabel(params.component_copy);
```

Comment:
not needed, IMO

Reply:
what about this:
![image](https://user-images.githubusercontent.com/1849576/168293147-270a08cc-f869-4f17-97a3-ace95d8d0982.png)
it may be problematic if we don't sanitize that `module_title` param, because it will be a copy, for example `"teléfono"`
![image](https://user-images.githubusercontent.com/1849576/168293309-4d9d7318-89ae-4e77-a450-fea83c01a3cc.png)